docs: 874 benchmarking (#944)
aneoconsulting/ArmoniK: test/test_plan.md

Reviewer: Need to be added to the `.docs` folder (a merge from main is required)

Reviewer: There is also missing informations about how to use and the purpose of each file (bench_10k, cleaner, test and wjson)!

 # Test plan of Armonik "2.12.0"
-[Introduction](#Introduction)

-[The product and the functionalities to test](#the-product-and-the-functionalities-to-test)

-[Prerequisite and exigences](#prerequisite-and-exigences)

-[The tools used to do the tests](#the-tools-used-to-do-the-tests)

-[Tests environment](#Tests-environment)

-[Exploited ressources](#exploited-ressources)

-[The estimated results](#the-estimated-results)

-[Tests to do](#tests-to-do)

# Introduction

> This test plan describes the performance tests of ArmoniK. The purpose of those tests is to mesure the performance of the differents functionnalities of ArmoniK(Tasks submission, Tasks treatement,...).For this, we will do different tests:

Reviewer: ```suggestion
This test plan describes the performance tests of ArmoniK. The purpose of those tests is to mesure the performance of the differents functionalities of ArmoniK (Tasks submission, Tasks treatment, ...). For this, we will do different tests:
```

- Measure the execution time of different numbers of submitted tasks and different size of the tasks.
- Test the strong scalability.

The purpose of those tests is to have a reference numbers and performance of the versions of ArmoniK.

> # The product and the functionalities to test

Reviewer: ```suggestion
# The product and the functionalities to test

```

> ## Products

Reviewer: ```suggestion
## Products

```

- Bench "0.11.4" , Stresstest "2.12.0" on ArmoniK "2.12.0"
- The submission of the tasks
- The processing of the tasks
- The retrieving of the results 

> # Prerequisite and exigences

Reviewer: Add space after each title.

| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 | 100 | 200m  | 1000m | 1000m | 2000m | 50m | 1000m | 100K | 10KB | 5 | 5 |  |  |  |  |  |  |  |  |
| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 | 1000m | 1000m | 2000m | 50m | 1000m | 100K | 100KB | 5 | 10 |  |  |  |  |  |  |  |  |
| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 | 1000m | 1000m | 2000m | 50m | 1000m | 1M | 1KB | 5 | 1 |  |  |  |  |  |  |  |  |
| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 | 100 | 200m  | 1000m | 1000m | 2000m | 50m | 1000m | 1M | 10KB | 5 | 5 |  |  |  |  |  |  |  |  |
| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 | 100 | 200m  | 1000m | 1000m | 2000m | 50m | 1000m | 1M | 100KB | 5 | 10 |  |  |  |  |  |  |  |  |

> ## Other tests : 

Reviewer: ```suggestion
## Other tests
```

# The tools used to do the tests
## Scripts bash
- To lunch a warming up run then a bunch of runs with the same parameters and store them in files
## Python:
- Clean the files where we stocked the data in readable json files.
- Read the data and calculate the median and the mean of each run bunch.
## Json :
- We stock the results of the performance tests in Json file wich we will store in a database.

# Tests environment
- ArmoniK deployed on AWS via WSL2
- Object storag: Redis / S3

# Exploited ressources
- Ressources AWS(eks "1.25","elasticache","amazonMQ")

# The estimated results
- More pods -> faster treatement 

# Tests to do
## Bench
- Scalability tests:
    - 10k tasks on : 100 pods, 1000 pods, and 10000 pods
    - 100k tasks on : 100 pods, 1000 pods, and 10000 pods
    - 1M tasks on : 100 pods, 1000 pods, and 10000 pods
    ## with
    - The tasks duration : 500 ms
    - The payload size : 100 KB
    - The result size : 100 KB

- Performance tests with a fix number of pods (100 pods) with and variable parameters:
    - The number of tasks
    - The tasks durations
    - The payload size
    - The result size

>     - 80k tasks, 1s per task, 1cpu per worker

Reviewer 2: It is 80k cores for 1 million tasks with 1k payload and 1k result

Author: fixed !

Reviewer 2: well, we would like it to run at least 10 minutes so it is more like 50 millions tasks

Author: done !
